libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -1147,6 +1155,47 @@ impl NodeAPI for Greenlight {
         payment.try_into()
     }
 
+    async fn send_trampoline_payment(
+        &self,
+        bolt11: String,
+        amount_msat: u64,
+        label: Option<String>,
+        trampoline_node_id: Vec<u8>,
+    ) -> NodeResult<Payment> {
+        let invoice = parse_invoice(&bolt11)?;
+        validate_network(invoice.clone(), self.sdk_config.network)?;
+        let label = serde_json::to_string(&PaymentLabel {
+            trampoline: true,
+            client_label: label,
+            unix_nano: SystemTime::now().duration_since(UNIX_EPOCH)?.as_nanos(),
+            amount_msat,
+        })?;
+        let mut client = self.get_client().await?;
+        let request = TrampolinePayRequest {
+            bolt11,
+            trampoline_node_id,
+            amount_msat,
+            label,
+            maxdelay: u32::default(),
+            description: String::default(),
+            maxfeepercent: f32::default(),
+        };
+        let result = self
+            .with_keep_alive(client.trampoline_pay(request))
+            .await?
+            .into_inner();
+
+        let client = self.get_node_client().await?;
+
+        // Before returning from send_payment we need to make sure it is
+        // persisted in the backend node. We do so by polling for the payment.
+        // TODO: Ensure this works with trampoline payments
+        // NOTE: If this doesn't work with trampoline payments, the sync also
+        // needs updating.
+        let payment = Self::fetch_outgoing_payment_with_retry(client, result.payment_hash).await?;
```

Comment:
About the comment, do we know if that works with trampoline?

Reply:
Not yet, for that we need the transport error to be resolved.